### DISCUSSION TITLE
Documentation for woof-CE development.

### DISCUSSION THREAD

**Discussion by lakshayrohila** [puppylinux-woof-CE/woof-CE#3557] [Ideas]
The current guide ([Wiki](https://github.com/puppylinux-woof-CE/woof-CE/wiki)) for **woof-CE** focuses more on building a puppy than improving woof-CE by code. Atleast those who can, can contribute code. But it would be possible only if they know how to contribute some. And what I mean by this is not creating a documentation which teaches using github/git or even programming, for there are documentations already for those. But instead, a documentation which explains which directories/files are there for what. Currently only `/woof-code/README.pkgs_specs` explains a bit about configuring the development for installation of desired programs, but not for the complete **woof-CE.** So why not provide a documentation for the 'development' part, if not there already, or else if present then provide more easy access to it.

I basically ask to provide a documentation that explains all the 'five' directories (or atleast `woof-*` ones) and their contents.

Being a 'newbie' to woof-CE, and having hard time merely guessing what goes for what, I can relate this need a lot with myself.

**Comment by dimkr** [comment]
What do you think of #3558?

**Reply by lakshayrohila** [reply]
Seems to be great!

**Reply by lakshayrohila** [reply]
Thanks!

**Reply by dimkr** [reply]
What else would you change or add?

**Reply by lakshayrohila** [reply]
Some different syntaxes for different files such as `/woof-code/rootfs-petbuilds/*/pet.specs` and `/woof-code/rootfs-packages/*/pet.specs` could be added.

**Reply by lakshayrohila** [reply]
Thinking more about this, maybe the syntaxes might not be good to be directly added to `README.md`, for I believe many those who are way far from programming may scare off reading that. Maybe a different file such as `DOCUMENTATION.md` might be a great place to put them in.

**Reply by dimkr** [reply]
How about GitHub wiki pages for these topics?

**Reply by lakshayrohila** [reply]
That would also be an option. I think that wiki might be a great place to put all the documentation part in, just like [conky](https://github.com/brndnmtthws/conky/wiki) has done.

**Comment by dimkr** [comment]
@lakshayrohila https://github.com/puppylinux-woof-CE/woof-CE/pull/3580

**Reply by lakshayrohila** [reply]
Great work @dimkr! Thanks!

**Reply by lakshayrohila** [reply]
@dimkr, just read https://github.com/puppylinux-woof-CE/woof-CE#building-packages-from-source. Just a request, could you please add the details for `pet.specs` file syntax? Its very non-intuitive to understand by taking a look at few different `pet.specs` files.

**Reply by dimkr** [reply]
pet.specs is needed only to register packages with PPM, so PPM considers them to be installed. Old .pet package or PPM documentation explains the pet.specs format.
